### ISSUE TITLE
Training Deep Neural Networks on Noisy Labels with Bootstrapping

### ISSUE THREAD

**Issue by leo-p** [leo-p/papers#38]
https://arxiv.org/pdf/1412.6596.pdf

Current state-of-the-art deep learning systems for visual object recognition and detection use purely supervised training with regularization such as dropout to avoid overfitting. The performance depends critically on the amount of labeled examples, and in current practice the labels are assumed to be unambiguous and accurate. However, this assumption often does not hold; e.g. in recognition, class labels may be missing; in detection, objects in the image may not be localized; and in general, the labeling may be subjective. In this work we propose a generic way to handle noisy and incomplete labeling by augmenting the prediction objective with a notion of consistency. We consider a prediction consistent if the same prediction is made given similar percepts, where the notion of similarity is between deep network features computed from the input data. In experiments we demonstrate that our approach yields substantial robustness to label noise on several datasets. On MNIST handwritten digits, we show that our model is robust to label corruption. On the Toronto Face Database, we show that our model handles well the case of subjective labels in emotion recognition, achieving state-of-the- art results, and can also benefit from unlabeled face images with no modification to our method. On the ILSVRC2014 detection challenge data, we show that our approach extends to very deep networks, high resolution images and structured outputs, and results in improved scalable detection.

**Comment by leo-p** [comment]
## Summary:
- *Dataset:* [MNIST](yann.lecun.com/exdb/mnist/), Toroto Faces Database, [ILSVRC2014](http://www.image-net.org/challenges/LSVRC/2014/).
- *Objective:* Design a loss to make deep network robust to label noise.

## Inner-workings:
Three types of losses are presented:
- reconstruciton loss:
<img width="454" alt="screen shot 2017-06-26 at 11 00 07 am" src="https://user-images.githubusercontent.com/17261080/27532200-bb42b8a6-5a5f-11e7-8c14-673958216bfc.png">

- soft bootstrapping which uses the predicted labels by the network `qk` and the user-provided labels `tk`:
<img width="308" alt="screen shot 2017-06-26 at 11 10 43 am" src="https://user-images.githubusercontent.com/17261080/27532296-1e01a420-5a60-11e7-9273-d1affb0d7c2e.png">

- hard bootstrapping replaces the soft predicted labels by their binary version:
<img width="317" alt="screen shot 2017-06-26 at 11 12 58 am" src="https://user-images.githubusercontent.com/17261080/27532439-a3f9dbd8-5a60-11e7-91a7-327efc748eae.png">
<img width="231" alt="screen shot 2017-06-26 at 11 13 05 am" src="https://user-images.githubusercontent.com/17261080/27532463-b52f4ab4-5a60-11e7-9aed-615109b61bd8.png">

## Architecture:
They test with Feed Forward Neural Networks only.

## Results:
They use only permutation noise with a very high probability compared with what we might encounter in real-life.

<img width="641" alt="screen shot 2017-06-26 at 11 29 05 am" src="https://user-images.githubusercontent.com/17261080/27533105-b051d366-5a62-11e7-95f3-168d0d2d7841.png">

The improvement for small noise probability (<10%) might not be that interesting.
